microsoft/testsuites/core/azure_image_standard.py

```diff
@@ -1159,14 +1178,29 @@ def verify_resource_disk_readme_file(self, node: RemoteNode) -> None:
         ),
     )
     def verify_resource_disk_file_system(self, node: RemoteNode) -> None:
-        resource_disk_mount_point = node.features[Disk].get_resource_disk_mount_point()
-        node.features[Disk].get_partition_with_mount_point(resource_disk_mount_point)
-        disk_info = node.tools[Lsblk].find_disk_by_mountpoint(resource_disk_mount_point)
-        for partition in disk_info.partitions:
-            # by default, resource disk comes with ntfs type
-            # waagent or cloud-init will format it unless there are some commands hung
-            # or interrupt
-            assert_that(
-                partition.fstype,
-                "Resource disk file system type should not equal to ntfs",
-            ).is_not_equal_to("ntfs")
+        node_disc = node.features[Disk]
+        resource_disks = node_disc.get_resource_disks()
+        if not resource_disks:
+            raise LisaException("Resource disk not found")
+        if "nvme" in resource_disks[0]:
```

Comment:
Wrap the repeat logic to reuse.

Reply:
Added a new method "get_resource_disk_type" for this. Which raises exception if no resource disk is available.